src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -3866,12 +3866,12 @@ public void M(ref Test obj)
                 // (6,9): error CS8373: The left-hand side of a ref assignment must be a ref variable.
                 //         this = ref this;
                 Diagnostic(ErrorCode.ERR_RefLocalOrParamExpected, "this").WithLocation(6, 9),
-                // (6,20): error CS1510: A ref or out value must be an assignable variable
+                // (6,20): error CS1605: Cannot use 'this' as a ref or out value because it is read-only
```

Comment:
>// (6,20): error CS1605: Cannot use 'this' as a ref or out value because it is read-only [](http://example.com/codeflow?start=16&length=88)

The change to diagnostics in this scenario feels unexpected.